packages/contracts/migrations/2_deploy_contracts.js

```diff
@@ -14,75 +14,45 @@ const VirtualAppAgreement = artifacts.require("VirtualAppAgreement");
 const ETHBalanceRefundApp = artifacts.require("ETHBalanceRefundApp");
 
 module.exports = (deployer, network) => {
-  deployer.deploy(Transfer).then(instance => {
-    deployer.link(Transfer, [VirtualAppAgreement, ConditionalTransaction]);
-    if (!tdr.isDryRunNetworkName(network)) {
-      return tdr.appendInstance(instance);
+  deployer.then(async () => {
+
+    const artifacts = [
+      {
+        'artifact': Transfer,
+        'dependents': [VirtualAppAgreement, ConditionalTransaction]
+      },
+      {
+        'artifact': StaticCall,
+        'dependents': [ConditionalTransaction, Conditional]
+      },
+      ConditionalTransaction,
+      VirtualAppAgreement,
+      MultiSend,
+      NonceRegistry,
+      PaymentApp,
+      ETHBalanceRefundApp,
+      ProxyFactory,
+      Registry,
+      Signatures,
+      // FIXME: This doesn't need to be deployed, but eth-gas-reporter breaks
+      // if it isn't deployed.
+      Conditional
+    ];
+
+    for (const entry of artifacts) {
+      const artifact = entry.artifact || entry;
+      const instance = await deployer.deploy(artifact);
+
+      // link each dependent contract individually
+      // https://github.com/trufflesuite/truffle/pull/1489
+      for (const dependent of entry.dependents || []) {
```

Comment:
```suggestion
      if (entry.dependents) {
          for (const dependent of entry.dependents) {
```
This is just my style preference, makes this code easier to skim.